### ISSUE TITLE
XAxis labels position

### ISSUE THREAD

**Issue by zykis** [danielgindi/Charts#2486]
I have a xaxis labels i want to be just underneath of bars in BarChartView
![cutmypic](https://cloud.githubusercontent.com/assets/5877248/26574999/35cec130-452c-11e7-8bef-644038785ffa.png)

But as you can see, there is a misplaced labels.
XAxisRenderer.swift:211 shows me the following xAxis.entries:
0 : 1495530709.7142859 - 23.05 09:11:49
1 : 1495631273.1428573 - 24.05 13:07:53 (27:56:04 diff)
2 : 1495731836.5714288 - 25.05 17:03:56 (27:56:03 diff)
3 : 1495832400.0000002 - 26.05 21:00:00 (27:56:04 diff)
4 : 1495932963.4285717 - 28.05 00:56:03 (27:56:04 diff)
5 : 1496033526.8571432 - 29.05 04:52:06 (27:56:03 diff)
6 : 1496134090.2857146 - 30.05 08:48:10 (27:56:04 diff)

XAxisRenderer.swift:211 shows me the following position.x:
0: 45.81
1: 78.81 (33 diff)
2: 121.85 (33 diff)
3: 169.89 (43 diff)
4: 207.92 (37 diff)
5: 250.96 (32 diff)
6: 283.89 (32 diff)

I guess the problem may be in 
XAxisRenderer.swift:200
let valueToPixelMatrix = transformer.valueToPixelMatrix

Any suggestions?
P.S.:

```
// BarChartView settings:
ChartXAxis* xaxis = _chartView.xAxis;
DateAxisValueFormatter* dateFormatter = [[DateAxisValueFormatter alloc] init];
xaxis.drawGridLinesEnabled = YES;
xaxis.drawAxisLineEnabled = YES;
xaxis.drawLabelsEnabled = YES;
xaxis.labelPosition = XAxisLabelPositionBottom;
xaxis.labelTextColor = [UIColor whiteColor];
xaxis.valueFormatter = dateFormatter;
xaxis.labelFont = [UIFont systemFontOfSize:7];
xaxis.avoidFirstLastClippingEnabled = YES;
xaxis.forceLabelsEnabled = YES;
xaxis.forceLabelsCount = 7;
```

```
// DateAxisValueFormatter.m
- (NSString*)stringForValue:(double)value axis:(ChartAxisBase *)axis {
    NSDate* date = [NSDate dateWithTimeIntervalSince1970:value];
    NSDateFormatter* toFormatter = [[NSDateFormatter alloc] init];
    [toFormatter setDateFormat:@"dd.MM"];
    NSString* res = [toFormatter stringFromDate:date];
    return res;
}
```

**Comment by liuxuan30** [comment]
have you rewrite any code?

**Comment by zykis** [comment]
Not yet.

**Comment by liuxuan30** [comment]
from your screenshot, it seems like the label is not aligned, so something wrong with your setup.

**Comment by zykis** [comment]
Yep, i just want them to be aligned properly. That is the problem

**Comment by liuxuan30** [comment]
but I suppose it should be your setup issue. ChartsDemo works fine and your bar chart looks normal

**Comment by zykis** [comment]
Ok, thanks!
